### PR TITLE
add DO persister tests, fix prefix parsing, fix empty storage handling

### PR DESCRIPTION
**Pull request by bitmage**
## Summary

  Fixes two bugs in the Durable Object Storage persister when using a non-empty storagePrefix:

  1. Prefix parsing bug (line 46): slice(key, storagePrefix.length, 1) returns empty string when prefix length > 1. Fixed to slice(key, storagePrefix.length, storagePrefix.length + 1).
  2. Empty storage handling (line 103-105): getPersisted() returned valid-but-empty content [[{}, '', 0], [{}, '', 0]] when storage was empty, causing load() to wipe existing store data. Now returns undefined when no data exists.

## How did you test this change?

There were formerly no tests (!) for this persistence provider.  I added a mock structure which allows it to be tested in a way that I believe is idiomatic with the other providers.  Initially the first bug was my target, but I discovered the second when I got the test suite to run.

### CODE REVIEW

File: src/persisters/persister-durable-object-storage/index.ts [hunk 1]
```diff
@@ -100,6 +100,9 @@ export const createDurableObjectStoragePersister = ((
               : 0,
         ),
     );
+    if (Object.keys(tables[0]).length === 0 && Object.keys(values[0]).length === 0) {
```
Comment (jamesgpearce) [inline src/persisters/persister-durable-object-storage/index.ts]: we have some utility functions for this but I can apply them
